clang/test/CIR/CodeGen/OpenCL/kernel-attributes.cl

```diff
@@ -0,0 +1,28 @@
+// RUN: %clang_cc1 -fclangir -emit-cir -triple x86_64-unknown-linux-gnu %s -o %t.cir
+// RUN: FileCheck %s --input-file=%t.cir --check-prefix=CIR
+// RUN: %clang_cc1 -fclangir -emit-llvm -triple x86_64-unknown-linux-gnu %s -o %t.ll
+// RUN: FileCheck %s --input-file=%t.ll --check-prefix=LLVM
+
+typedef unsigned int uint4 __attribute__((ext_vector_type(4)));
+
+kernel  __attribute__((vec_type_hint(int))) __attribute__((reqd_work_group_size(1,2,4))) void kernel1(int a) {}
+kernel __attribute__((vec_type_hint(uint4))) __attribute__((work_group_size_hint(8,16,32))) void kernel2(int a) {}
+kernel __attribute__((intel_reqd_sub_group_size(8))) void kernel3(int a) {}
+
+// CIR: #fn_attr[[KERNEL3:[0-9]*]] = {{.+}}ocl.kernel_metadata = #cir.ocl.kernel_metadata<intelReqdSubGroupSize = 8 : i32>{{.+}}
```

Comment:
Please also try to put this together with the matching source declarations, easier to read, you could use `CIR-DAG` if needed.